File: antd-dart/example/06_private_data.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:antd/antd.dart';

/// Demonstrates storing and retrieving private (encrypted) data.
void main() async {
  final client = AntdClient();

  try {
    // Store private data
    final data = Uint8List.fromList(utf8.encode('my secret data'));
    final result = await client.dataPutPrivate(data);
    print('Private data stored (cost: ${result.cost} atto)');
    print('Data map: ${result.address}');

    // Retrieve private data using the data map
    final retrieved = await client.dataGetPrivate(result.address);
    print('Retrieved: ${utf8.decode(retrieved)}');
  } on AntdError catch (e) {
    print('Error: $e');
  } finally {
    client.close();
  }
}

File: antd-rust/examples/06-private-data.rs
use antd_client::{Client, DEFAULT_BASE_URL};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new(DEFAULT_BASE_URL);

    // Store private (encrypted) data
    let result = client.data_put_private(b"secret payload", None).await?;
    println!("Private data stored (cost: {} atto)", result.cost);
    println!("Data map: {}", result.address);

    // Retrieve private data using the data map
    let data = client.data_get_private(&result.address).await?;
    println!("Decrypted: {}", String::from_utf8_lossy(&data));

    Ok(())
}
